### ISSUE TITLE
Frequent terms 

### ISSUE THREAD

**Issue by lisaoshita** [shannonpileggi/SR--Oshita--iFixit#11]
I’m still kind of struggling to understand what we’re doing with the frequent terms. Just to make sure, we’re trying to find the most frequently used terms among unanswered and answered questions. But since there was quite a bit of overlap between the most frequently used terms for answered/unanswered questions (ex: screen, phone), we need to find words that are unique to each category. So we’re calculating the proportion of times each term shows up in it’s answered/unanswered category, then finding the ratio to see which group (answered/unanswered) the word is most likely to belong to. To pick out the words I’ll use, I need to determine min and max proportion thresholds, so that any words between those proportions is what I’ll include as most frequently used. And to determine those thresholds, it’s kind of up to however I see fit based off of the ratios and the distribution of the proportions? I feel like I'm almost understanding this- I just keep losing it lol

**Comment by anthonypileggi** [comment]
Yup, we are trying to identify words that are present at very different rates for an answered vs unanswered question.  So the interpretation works something like given that a post contains "iphone", it is 8 times more likely to have an answer.

We noticed that some words show up a lot, or not enough -- that's why we are considering the min/max thresholds.  Even if a word is incredibly predictive (e.g. x25 more likely), if it only shows up in 1/1,000,000 posts then it is rarely going to help us make decisions.  So our goal is to find a set of words that shows up frequently enough to be useful as a predictor.

**Comment by shannonpileggi** [comment]
Yes, this makes sense to me.  Suppose the the word "phone" is used in 80% of answered and 80% of unanswered questions - then it has a ratio of 1, and won't be a valuable predictor.  What about if the word "gizmo" is used in 1% of answered and 2% of unanswered?  Then it has a ratio of 1/2, and could be a strong predictor.  However, it isn't used frequently so it might not be helpful (and could lead to large standard errors in parameter estimates).

**Comment by lisaoshita** [comment]
Ok! That all makes sense. Now I'm kind of struggling with how to approach this. For unanswered questions, I was thinking about setting my minimum threshold to 1%, which only gives me 28 terms. Then I was thinking that I would order those terms according to the ratio from lowest to highest, and grab the terms with the lowest ratios? Since my ratio = proportion answered/proportion unanswered, a ratio of less than 1 would mean that the term is more likely to be in an unaswered question? If I subset to all terms that have a ratio of less than 1, that gives me only 9 terms. I was also thinking that I might not need a max threshold?

**Comment by lisaoshita** [comment]
Also, I'm wondering if I need to worry about any confounding that might be going on here? For example, I saw that the term camera occurs in 2% of unanswered is more likely to be in unanswered questions (ratio = 0.6). But earlier I also found that the camera category had the highest average/median time until answer values, so it would make sense that the term would be in unanswered questions. Wondering if I should be worried about overlap like this?

**Comment by shannonpileggi** [comment]
Ratios of more than one might also be of interest, too.  This is because that means those words would be more likely to be answered.

If you want to discuss the possible confounding stuff more, I think that conversation is easiest to have in person (but I think we got to it on thursday's meeting).

**Comment by lisaoshita** [comment]
Ok! For the terms I’m considering important among unanswered questions, I put the minimum proportion as 0.01, and the ratio threshold set to less than 1, since the ratio = proportion of times it appears in answered questions/proportion of times it appears in unanswered questions. As for the terms I’m considering in answered questions, I kept the minimum proportion the same at 0.01, but set the ratio threshold as greater than 1. 

Also, I did what we talked about yesterday and narrowed the list of frequent terms for answered and unanswered questions to just terms regarding devices (I took out words like "keep" and "turning"), and there is still a significant difference in survival times for questions that contain those words and questions without.
